### PR TITLE
Dump from S3 to ODC DB

### PR DESCRIPTION
**Pull request by whatnick**
Initial cut of reading data from S3 bucket and putting it into an index chaining pieces of odc-tools code together.

### CODE REVIEW

File: odc_index/s3_to_dc.py [hunk 1]
```diff
@@ -3,9 +3,39 @@
...
+import logging
 
 import click
-from odc.aio import s3_find_glob
+from odc.aio import s3_find_glob, S3Fetcher
+from odc.index import from_yaml_doc_stream
+from datacube import Datacube
+
+
+def dump_to_odc(data_stream , dc : Datacube) -> tuple:
```
Comment (Kirill888) [inline odc_index/s3_to_dc.py:14]: it's 
```python
from typing import Tuple

def dump_to_odc(data_stream , dc : Datacube) -> Tuple[int,int]:
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -53,4 +53,6 @@ jobs:
         docker-compose run dc-index datacube metadata add https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/eo3/eo3.odc-type.yaml
         docker-compose run dc-index datacube product add https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/products/cemp_insar_alos_displacement.yaml
         docker-compose run dc-index s3-to-dc s3://dea-public-data/cemp_insar/insar/displacement/alos//**/*.yaml cemp_insar_alos_displacement
+        echo "Checking Indexed Datasets Count"
+        docker exec datacube-index_dc-db_1 psql -U postgres -c "SELECT count(*) from agdc.dataset"
```
Comment (alexgleith) [inline .github/workflows/main.yml:57]: Why docker and not docker-compose exec?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -53,4 +53,6 @@ jobs:
         docker-compose run dc-index datacube metadata add https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/eo3/eo3.odc-type.yaml
         docker-compose run dc-index datacube product add https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/develop/digitalearthau/config/products/cemp_insar_alos_displacement.yaml
```
Comment (alexgleith) [inline .github/workflows/main.yml:54]: Also why not use exec here? run will create a new container. (Maybe the container isn't running currently, I guess)

Comment (whatnick) [reply .github/workflows/main.yml:54]: Yes not running , this wraps around tools rather than a web-app.

**Pull request merged**